src/net.h

```diff
@@ -644,20 +644,40 @@ class CNetMessage {
  * transport protocol agnostic CNetMessage (command & payload)
  */
 class TransportDeserializer {
+public:
+    // prepare for next message
+    virtual void Reset() = 0;
+    // returns true if the current deserialization is complete
+    virtual bool Complete() const = 0;
+    // checks if the potential message in deserialization is oversized
+    virtual bool OversizedMessageDetected() const = 0;
+    // set the serialization context version
+    virtual void SetVersion(int nVersionIn) = 0;
```

Comment:
b8e9633d127fd1e740b3c9e25a6ed5d480a77f25

nit, use correct style in new code? `int version`.